src/main/java/frc/robot/subsystems/turretSubsystem/TurretCommandFactory.java

```diff
@@ -100,28 +135,111 @@ public Command zeroTurretCommand() {
         .asProxy();
   }
 
-  public Command moveUpwards() {
-    if (subsystem == null) return Commands.none();
-    return subsystem.run(() -> subsystem.moveUp()).withName("moveShooterUp").asProxy();
-  }
-
   public Command getAimTurretCommand() {
     if (subsystem == null) return new StartEndCommand(() -> {}, () -> {});
-    return subsystem.run(() -> subsystem.aimTurret()).withName("AimTurretCommand").asProxy();
+    return Commands.either(shortRangeShot(), longRangeShot(), () -> false);
+  }
+
+  public Command shortRangeShot() {
+    return subsystem.startEnd(
+        () -> {
+          subsystem.setTurretPos(Math.toRadians(00));
+          subsystem.setPitchPos(Math.toRadians(37));
+        },
+        () -> {});
+  }
+
+  public Command longRangeShot() {
+    return subsystem.run(
+        () -> {
+          if (Constants.enabledSubsystems.turretRotationEnabled) {
+            visionTracking();
+          }
+          if (Constants.enabledSubsystems.turretPitchEnabled) {
+            double distance = distanceEstimateMeters();
+            limelightDistance.log(distance);
+            subsystem.setPitchPos(pitchMap.get(distance));
+          }
+        });
+  }
+
+  private void visionTracking() {
+    visionTracking(() -> odometryPointing());
+  }
+
+  private void visionTracking(Supplier<Rotation2d> searchingBehavior) {
+    visionTracking(getSpeakerTag(), searchingBehavior);
+  }
+
+  private int getSpeakerTag() {
+    return RSM.getAllianceColor() == AllianceColor.RED
+        ? LimelightConstants.SPEAKER_TAG_ID_RED
+        : LimelightConstants.SPEAKER_TAG_ID_BLUE;
+  }
+
+  private double distanceEstimateMeters() {
+    double visionDistance = vision.getDistance(getSpeakerTag());
+    if (Double.isNaN(visionDistance)) {
+      return odometryDistance();
+    }
+    return visionDistance;
+  }
+
+  private double odometryDistance() {
+    return positionRelativeToSpeaker().getNorm();
+  }
+
+  DebugEntry<Double> visionRotation;
+
+  /**
+   * @param targetTag the tag to search towards
+   * @param searchingBehavior target rotation in degrees
+   */
+  private void visionTracking(int targetTag, Supplier<Rotation2d> searchingBehavior) {
+    subsystem.setPositionErrorSupplier(
+        () -> {
+          double errDegrees = vision.getTurretYaw(targetTag);
+          if (Double.isNaN(errDegrees)) {
+            return searchingBehavior.get().getRotations() + subsystem.getTurretPos();
+          }
+          double err = Units.degreesToRotations(errDegrees - 4);
```

Comment:
magic number?  Are we getting a consistent error from photonvision?  Preferable to remove this.  But if not, add a constant